WordPressAuthenticatorTests/Credentials/CredentialsTests.swift

```diff
@@ -0,0 +1,127 @@
+import XCTest
+@testable import WordPressAuthenticator
+
+class CredentialsTests: XCTestCase {
+    
+    let token = "arstdhneio123456789qwfpgjluy"
+    let siteURL = "https://example.com"
+    let username = "user123"
+    let password = "arstdhneio"
+    let xmlrpc = "https://example.com/xmlrpc.php"
+    
+    
+    func testWordpressComCredentialsInit() {
+        let wpcomCredentials = WordPressComCredentials(authToken: token,
+                                                       isJetpackLogin: false,
+                                                       multifactor: false,
+                                                       siteURL: siteURL)
+        
+        XCTAssertEqual(wpcomCredentials.authToken, token)
+        XCTAssertEqual(wpcomCredentials.isJetpackLogin, false)
+        XCTAssertEqual(wpcomCredentials.multifactor, false)
+        XCTAssertEqual(wpcomCredentials.siteURL, siteURL)
+    }
+    
+    func testWordPressComCredentialsSiteURLReturnsDefaultValue() {
+        let wpcomCredentials = WordPressComCredentials(authToken: token,
+                                                       isJetpackLogin: false,
+                                                       multifactor: false,
+                                                       siteURL: "")
+        
+        let expected = "https://wordpress.com"
+    
+        XCTAssertEqual(wpcomCredentials.siteURL, expected)
+    }
+    
+    func testWordPressComCredentialsEquatable() {
+        let lhs = WordPressComCredentials(authToken: token,
+                                          isJetpackLogin: false,
+                                          multifactor: false,
+                                          siteURL: siteURL)
+        
+        let rhs = WordPressComCredentials(authToken: token,
+                                          isJetpackLogin: false,
+                                          multifactor: false,
+                                          siteURL: siteURL)
+        
+        XCTAssertTrue(lhs == rhs)
```

Comment:
It's great that we have a test for the custom `==` implementation.

But since the implementation is custom, what's interesting to test is that it returns `true` in those cases were the instances don't have all the same values, but only those that matter to our implementation.

https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Credentials/WordPressComCredentials.swift#L39-L41

That's a point worth generalizing: tests should assert interesting behaviors of the software. Tests are a conversation between the developer and the codebase, the should ask questions the answers to which generate insight in the behavior of the software.